docs/websites/apache-tips-and-tricks/apache-new-file-handler.md

```diff
@@ -0,0 +1,589 @@
+files with a certain extension are called.  I could not find any help
+doing this online, despite how easy it turned out to be.  I have written
+this guide to help others succeed where I struggled to understand.
+
+This guide will use a simple CGI script to demonstrate how to configure
+Apache to use the script as a file handler when an appropriate file is
+called.  Then it will show how to convert the CGI script to a FastCGI
+script for performance reasons, and how to modify the configuration to
+use the FastCGI version instead of the CGI script.
+
+The main sections are arranged thus:
+- [Prerequisites](#Prerequisites)
+- [The CGI Script](#TheCGIScript)
+- [Configuring Apache to Use a File Handler](#ConfiguringApache)
+- [Test CGI File Handler](#TestCGIHandler)
+- [Upgrading CGI to FastCGI](#UpgradeToFastCGI)
+- [Test FastCGI File Handler](#TestFastCGIHandler)
+- [Reading the Target File](#ReadingTheTargetFile)
+- [CGI/FastCGI Programming Suggestions](#ScriptProgrammingSuggestions)
+
+## Prerequisites                            {#Prerequisites}
+
+The first three steps prepare the environment for running a CGI file handler,
+the remaining steps add the resources necessary for building and running a
+FastCGI program.
+
+### CGI Setup
+
+1. A running LAMP installation.  Complete the
+   [Getting Started](/docs/getting-started) guide.
+
+2. Enable the `actions` mod for configuring a custom file handler and restart
+   Apache
+
+       sudo a2enmod actions
+       sudo /etc/init.d/apache2 restart
+
+3. Install `build-essential` tools for for compiling the C-language example CGI
+   and FastCGI programs and the FastCGI libraries.
+   
+       sudo apt-get install build-essential
+
+### FastCGI Setup                           {#FastCGISetup}
+
+4. Download, compile, and install FastCGI libraries.
+
+   Note: When writing this guide, the otherwise lost contents of fastcgi.com have
+   been archived at [FastCGI Archives](https://github.com/FastCGI-Archives/FastCGI.com),
+   and the `wget` URL below reflects this relocation.
+
+   ~~~
+   cd /usr/local/src
+   sudo wget https://github.com/FastCGI-Archives/FastCGI.com/blob/master/original_snapshot/fcgi-2.4.1-SNAP-0910052249.tar.gz
+   sudo tar -xvz fcgi-2.4.1*
+   cd fcgi-2.4.1*
+   sudo ./configure
+   sudo make
+   sudo make install
+   ~~~
+
+5. Install and enable `mod_fastcgi`, then restart Apache
+
+       sudo apt-get install libapache2_mod_fastcgi
+       sudo a2enmod fastcgi
+       sudo /etc/init.d/apache2 restart
+
+
+## The CGI Script                           {#TheCGIScript}
+
+Many languages can be used to program a CGI script.  The example in this
+guide is a C program because I prefer that language, but it could as easily
+be written in Perl, Python, Java, or BASH.
+
+The example CGI script does nothing except announce its presence.
+
+1. Put the following source code into a file for compilation:
+
+   {:file} simplescript.c
```

Comment:
Incorrect callout. Check the [formatting guide](https://www.linode.com/docs/linode-writers-formatting-guide#files-and-file-excerpts)